src/OpenTelemetry.AutoInstrumentation/HeadersCapture/HeaderEnrichment.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Specialized;
+using System.Diagnostics;
+using System.Net.Http.Headers;
+#if NET6_0_OR_GREATER
+using Microsoft.AspNetCore.Http;
+#endif
+
+namespace OpenTelemetry.AutoInstrumentation.HeadersCapture;
+
+internal static class HeaderEnrichment
```

Comment:
Might use different name, to me it sounds more like it's enriching headers instead of activities with headers.

Maybe something like: "ActivityExtensions.EnrichWithHeaders" ?

Reply:
4037c6ded4e7fcb8c375bbf601b5eaa006bda361